deltacat/tests/compute/compactor_v2/test_compaction_session.py

```diff
@@ -11,80 +15,241 @@
         # setup
-        rcf_url.read_round_completion_file.return_value = None
         staged_source = stage_partition_from_file_paths(
-            self.NAMESPACE, ["test"], **self.deltacat_storage_kwargs
+            self.NAMESPACE, ["source"], **local_deltacat_storage_kwargs
         )
-        source_partition = ds.commit_partition(
-            staged_source, **self.deltacat_storage_kwargs
+
+        source_delta = commit_delta_to_staged_partition(
+            staged_source, [self.BACKFILL_FILE_PATH], **local_deltacat_storage_kwargs
         )
 
         staged_dest = stage_partition_from_file_paths(
-            self.NAMESPACE, ["destination"], **self.deltacat_storage_kwargs
+            self.NAMESPACE, ["destination"], **local_deltacat_storage_kwargs
         )
         dest_partition = ds.commit_partition(
-            staged_dest, **self.deltacat_storage_kwargs
+            staged_dest, **local_deltacat_storage_kwargs
         )
 
         # action
         rcf_url = compact_partition(
             CompactPartitionParams.of(
                 {
-                    "compaction_artifact_s3_bucket": "test_bucket",
+                    "compaction_artifact_s3_bucket": TEST_S3_RCF_BUCKET_NAME,
                     "compacted_file_content_type": ContentType.PARQUET,
                     "dd_max_parallelism_ratio": 1.0,
                     "deltacat_storage": ds,
-                    "deltacat_storage_kwargs": self.deltacat_storage_kwargs,
+                    "deltacat_storage_kwargs": local_deltacat_storage_kwargs,
                     "destination_partition_locator": dest_partition.locator,
                     "drop_duplicates": True,
                     "hash_bucket_count": 1,
-                    "last_stream_position_to_compact": source_partition.stream_position,
+                    "last_stream_position_to_compact": source_delta.stream_position,
                     "list_deltas_kwargs": {
-                        **self.deltacat_storage_kwargs,
+                        **local_deltacat_storage_kwargs,
                         **{"equivalent_table_types": []},
                     },
                     "primary_keys": [],
+                    "rebase_source_partition_locator": source_delta.partition_locator,
+                    "rebase_source_partition_high_watermark": None,
+                    "records_per_compacted_file": 4000,
+                    "s3_client_kwargs": {},
+                    "source_partition_locator": source_delta.partition_locator,
+                }
+            )
+        )
+
+        bucket, backfill_key1, backfill_key2 = rcf_url.strip("s3://").split("/")
+        assert bucket == TEST_S3_RCF_BUCKET_NAME
+
+        # Now delete the RCF at new location and copy it to old location
+        # Copy the RCF from rcf_url to another location
+        s3_resource.Object(TEST_S3_RCF_BUCKET_NAME, f"{backfill_key1}.json").copy_from(
```

Comment:
Which scenario will result in this copy behavior in reality? With the new changes, we will be both read/write from s3://source_partition_locator/destination_partition_locator?

Reply:
The copy is not done anywhere. This is to simulate the RCFs were written by previous commits.